src/test/java/guitests/FilterTextFieldTest.java

```diff
@@ -121,6 +62,96 @@ public void cancelTest() {
         awaitCondition(() -> toggle.get() % 2 == 1);
     }
 
+    @Test
+    public void filterTextFieldColor_validFilter_validFilterStyleApplied(){
+        FilterTextField field = getFirstPanelField();
+
+        clearField();
+        type("is:open");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+
+        clearField();
+        type("-is:open");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+
+        clearField();
+        type("hello");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+
+        clearField();
+        type("is:open & has:assignee");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+
+        clearField();
+        type("(is:closed)");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+    }
+
+    @Test
+    public void filterTextFieldColor_invalidFilter_invalidFilterStyleApplied(){
+        FilterTextField field = getFirstPanelField();
+
+        // Tests if parse errors applies invalid filter style to text field
+        clearField();
+        type("is:-open");
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+
+        clearField();
+        type("--");
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+
+        clearField();
+        type("is:-open & has:assignee");
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+
+        clearField();
+        type("(is:closed");
```

Comment:
Is this the same as just `(`? It helps if we can keep the examples minimal, so it's clear what the problem is.